### PR TITLE
Adds callback to report the download and upload progress

### PR DESCRIPTION
**Pull request by crsib**
This is required for the GSoC project on Source Separation to provide a better UX.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"

### CODE REVIEW

File: libraries/lib-network-manager/curl/CurlResponse.cpp [hunk 1]
```diff
@@ -378,5 +397,20 @@ size_t CurlResponse::HeaderCallback (const char* buffer, size_t size, size_t nit
...
+   CurlResponse* clientp, curl_off_t dltotal, curl_off_t dlnow,
+   curl_off_t ultotal, curl_off_t ulnow) noexcept
+{
+    std::lock_guard<std::mutex> callbackLock(clientp->mCallbackMutex);
+
+    if (dltotal > 0 && clientp->mDownloadProgressCallback)
+       clientp->mDownloadProgressCallback(dlnow, dltotal);
+
+    if (ultotal > 0 && clientp->mUploadProgressCallback)
+       clientp->mUploadProgressCallback(dlnow, dltotal);
```
Comment (Paul-Licameli) [inline libraries/lib-network-manager/curl/CurlResponse.cpp:410]: Did you mean `ulnow, ultotal`

**Review by Paul-Licameli — CHANGES_REQUESTED** [review]
One line looks like a mistake, and also the macOS build fails.

File: libraries/lib-network-manager/curl/CurlResponse.cpp (hunk 1 above)
Comment (crsib) [reply libraries/lib-network-manager/curl/CurlResponse.cpp]: Yes, thank you! That is fixed now

**Review by Paul-Licameli — APPROVED** [review]
That was the only issue in my first review.

**Pull request merged**